### PR TITLE
[FE] Add unified alert filters

### LINKED ISSUE #2022: Update Filter for Unified Alerts Page
### Description

With policy based alerts, rule match alerts, and rule errors alerts being merged into the same view in Unified Alerts page, we need to update our filter to be able to filter on all policy and rule based alerts

### Designs

https://app.abstract.com/projects/558a8de6-7134-4c8b-91c5-62074bb1279b/branches/df9d59c8-eeb1-452d-a5f5-3712b64010be/collections/92f16a44-b3f2-44ff-a132-5198d0673377

### Acceptance Criteria
- Ability to filter on rule ID and policy ID
- Ability to filter on Alert Type (policy fail, rule match, rule error)
- Ability to filter on severity
- Ability to filter  on status
- Ability to filter on resource type
- Ability to filter on Event quantity

### PR DESCRIPTION
**Pull request by 3nvi**
## Background

This PR modifies the filters present in the alert listing page, in order to add filters related to to policy-based alerts. This involves moving the log-types filter within the dropdown, as well as adding a few additional filters.

The final outcome is an alert listing page where both rule-based and policy-based alerts can be displayed and filtered.

Closes #2022 

## Designs

(We have agreed not to use a slider  since it doesn't allow you to specify `null` or `undefined` values) 

https://app.abstract.com/projects/558a8de6-7134-4c8b-91c5-62074bb1279b/branches/df9d59c8-eeb1-452d-a5f5-3712b64010be/commits/b4ce187f398f8b447052f91b0b8873117fbacbe7/files/7b054336-a10e-4814-b79a-80101efafbe6/layers/AA5EC271-155F-4BE3-8DAF-DC001026017D?collectionId=92f16a44-b3f2-44ff-a132-5198d0673377&collectionLayerId=7c91de43-7f2c-4fd1-89f9-f9c58308fa5e&mode=build&selected=root-AA5EC271-155F-4BE3-8DAF-DC001026017D

## Screenshots

Screen Shot 2020-12-09 at 5 05 02 PM

## Changes

- Move log types filter from the breadcrumbs to the dropdown & allow users to specify multiple values
- Add alert types & resource types filters in dropdown
- Allow dynamic rendering evaluation of the proper AlertCard, based on the type of Alert returned from the API
- Update & Enhance tests to cover the additional logic
- Modify `GenericItemCard.Value` positioning to account for some edge-case spacing issues 

## Testing

- `npm run test`

### CODE REVIEW

File: web/src/pages/ListAlerts/ListAlertBreadcrumbFilters/ListAlertBreadcrumbFilters.tsx [hunk 1]
```diff
@@ -17,101 +17,56 @@
...
+import { Flex } from 'pouncejs';
 import useRequestParamsWithoutPagination from 'Hooks/useRequestParamsWithoutPagination';
 
 import pick from 'lodash/pick';
 
 import FormikDateRangeInput from 'Components/fields/DateRangeInput';
-import FormikCombobox from 'Components/fields/ComboBox';
 import FormikAutosave from 'Components/utils/Autosave';
 import Breadcrumbs from 'Components/Breadcrumbs';
-import { useListAvailableLogTypes } from 'Source/graphql/queries';
```
Comment (alexmylonas) [inline web/src/pages/ListAlerts/ListAlertBreadcrumbFilters/ListAlertBreadcrumbFilters.tsx:31]: Since we are removing this, we could also exclude it from the Navigation
Although `ListRulesBreadcrumbFilters` is still using this.

I am inclining to disregard this but i would like your thought

**Review by alexmylonas — APPROVED** [review]
LGTM! I would like for @vorillaz to also have a look but I feel it's solid.

File: web/src/pages/ListAlerts/ListAlertBreadcrumbFilters/ListAlertBreadcrumbFilters.tsx (hunk 1 above)
Comment (3nvi) [reply web/src/pages/ListAlerts/ListAlertBreadcrumbFilters/ListAlertBreadcrumbFilters.tsx:31]: We are removing it from here and putting it in another component. We still need  the preloading in the navigation 😉

File: web/src/pages/ListAlerts/ListAlertFilters/DropdownFilters.tsx [hunk 2]
```diff
@@ -119,26 +143,51 @@ const DropdownFilters: React.FC = () => {
...
-                        min={0}
-                        label="Min Events"
-                        data-testid="alert-listing-min-event"
-                        placeholder="Minimum number of events"
+                      <Field
+                        as={FormikMultiCombobox}
+                        label="Log Types"
+                        name="logTypes"
+                        placeholder="Select log types"
+                        items={logTypeData?.listAvailableLogTypes?.logTypes ?? []}
```
Comment (vorillaz) [inline web/src/pages/ListAlerts/ListAlertFilters/DropdownFilters.tsx:151]: Nice one since we now can have an option to delete Log types on demand.

File: web/src/pages/ListRules/ListRules.test.tsx [hunk 3]
```diff
@@ -397,7 +397,7 @@ describe('ListRules', () => {
 
     // Expect the sort dropdown to be empty by default. Empty = "All Types" for this filter.
     const logTypesFilter = getAllByLabelText('Log Type')[0];
-    expect(logTypesFilter).toHaveValue('All types');
+    expect(logTypesFilter).toHaveValue('All Types');
```
Comment (vorillaz) [inline web/src/pages/ListRules/ListRules.test.tsx:400]: I am a bit confused, since we are using a multi combobox this assertion should get removed.
`All Types` was used since the first implementation was consuming a dropdown component.
Are we using a different component for Rules' filtering?

**Review by vorillaz — APPROVED** [review]
LGTM, thanks @3nvi for this one.

File: web/src/pages/ListRules/ListRules.test.tsx (hunk 3 above)
Comment (3nvi) [reply web/src/pages/ListRules/ListRules.test.tsx:400]: Yeah we moved the `log types` from breadcrumbs -> dropdown in the alerts page, while making them accept multiple values.

They are still accepting a single value for the Rules page, since nothing changed there ( [related designs](https://app.abstract.com/projects/558a8de6-7134-4c8b-91c5-62074bb1279b/branches/0d293ff7-a6c6-4ddc-9dd0-442c63b91995/commits/f467dd3cbee270b2c511b990ea9e982b1fccc481/files/c2bbc1a8-4778-4cd6-93a5-4630d8868201/layers/C33133F6-EFC3-41AF-9175-5DF64C2EBD72) )
